### DISCUSSION TITLE
Getting timeout, when the response takes >1 minute

### DISCUSSION THREAD

**Discussion by ravi160822** [fastapi/fastapi#9376] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/test_time")
async def test_time():
    time.sleep(65)
    return 1
```

### Description

<html><body><h1>504 Gateway Time-out</h1>
The server didn't respond in time.
</body></html>

I get 504 response code

### Operating System

Linux

### Operating System Details

This works on local, but I get error when I deploy it to dev branch
I set this also,

        uvicorn.run(app, host=host, port=port, timeout_keep_alive=10000)

but still it gives error

### FastAPI Version

0.88.0

### Python Version

3.10

**Comment by Viicos** [comment]
The `timeout_keep_alive` isn't related to your task (see [here](https://stackoverflow.com/questions/73152924/uvicorn-server-timeout#comment129197767_73152924)). I'm not sure this is related to `FastAPI` or `uvicorn`, as a 504 error happens generally when a proxy between your client and your `FastAPI` instance has timed out

**Comment by ravi160822** [comment]
it does not time out on local, but it timeout after 60 seconds on dev, so any api which takes >60 sec would give 504 error as response, do I need to make changes to the docker file, to increase this timeout value. I have not specified it anywhere.

**Reply by Viicos** [reply]
Are you using any kind of web server/reverse proxy such as `nginx` on dev?

**Reply by ravi160822** [reply]
yes there is nginx mentioned on the kustomization file in Kubernetes

**Reply by Viicos** [reply]
Then it might be `nginx` that is responsible for the timeout (default timeout is indeed 60 seconds). You'll have to configure it properly using `proxy_read_timeout`/`proxy_connect_timeout`/`proxy_send_timeout`

**Reply by yuenherny** [reply]
I had the same issue on my app deployed on Kubernetes and add these to `ingress.yaml` helped:
```
annotations:
    nginx.ingress.kubernetes.io/proxy-connect-timeout: "<seconds>"
    nginx.ingress.kubernetes.io/proxy-read-timeout: "<seconds>"
    nginx.ingress.kubernetes.io/proxy-send-timeout: "<seconds>"
    nginx.ingress.kubernetes.io/keep-alive-timeout: "<seconds>"
```

**Comment by YuriiMotov** [comment, marked as answer]
Don't use sync blocking code (`time.sleep(65)`) inside `async` functioins!
